test/stress/thousand_namespaces_test.go

```diff
@@ -0,0 +1,125 @@
+
+	resources := []*unstructured.Unstructured{
+		e2eutil.ManifestToUnstructured([]byte(cronTabCRDYaml)),
+	}
+
+	for i := 1; i <= 1000; i++ {
+		ns := fmt.Sprintf("%s-%d", namespaceName, i)
+		namespaceObj := e2eutil.TemplateToUnstructured(namespaceTemplate, struct{ Namespace string }{
+			Namespace: ns,
+		})
+		resources = append(resources, namespaceObj)
+
+		configMapObj := e2eutil.TemplateToUnstructured(configMapTemplate, struct{ Name, Namespace string }{
+			Name:      "cm-1",
+			Namespace: ns,
+		})
+		resources = append(resources, configMapObj)
+
+		cronTabObj := e2eutil.TemplateToUnstructured(cronTabTemplate, struct{ Name, Namespace string }{
+			Name:      "cm-1",
+			Namespace: ns,
+		})
+		resources = append(resources, cronTabObj)
+	}
+
+	defer func() {
+		// delete in reverse order (namespaces last)
+		for i := len(resources) - 1; i >= 0; i-- {
+			e2eutil.DeleteUnstructuredIfExists(ctx, c, resources[i])
+		}
+	}()
+
+	start := time.Now()
+
+	applierEvents := e2eutil.RunCollect(applier.Run(ctx, inventoryInfo, resources, apply.ApplierOptions{
+		// SSA reduces GET+PATCH to just PATCH, which is faster
+		ServerSideOptions: common.ServerSideOptions{
+			ServerSideApply: true,
+			ForceConflicts:  true,
+			FieldManager:    "cli-utils.kubernetes.io",
+		},
+		ReconcileTimeout: 30 * time.Minute,
+		EmitStatusEvents: false,
+	}))
+
+	duration := time.Since(start)
+	klog.Infof("Applier.Run execution time: %v", duration)
+
+	for _, e := range applierEvents {
+		Expect(e.ErrorEvent.Err).To(BeNil())
+	}
+	for _, e := range applierEvents {
+		Expect(e.ApplyEvent.Error).To(BeNil(), "ApplyEvent: %v", e.ApplyEvent)
+	}
+	for _, e := range applierEvents {
+		if e.Type == event.WaitType {
+			Expect(e.WaitEvent.Operation).To(BeElementOf(event.ReconcilePending, event.Reconciled), "WaitEvent: %v", e.WaitEvent)
+		}
+	}
+
+	// By("Verify 1000 ConfigMaps created")
+	// assertUnstructuredCount(ctx, c, deployment1Obj, 1000)
+
+	By("Verify inventory")
+	invConfig.InvSizeVerifyFunc(ctx, c, inventoryName, namespaceName, inventoryID, len(resources), len(resources))
+
+	By("Destroy LOTS of resources")
+	destroyer := invConfig.DestroyerFactoryFunc()
+
+	start = time.Now()
+
+	destroyerEvents := e2eutil.RunCollect(destroyer.Run(ctx, inventoryInfo, apply.DestroyerOptions{
+		InventoryPolicy: inventory.PolicyAdoptIfNoInventory,
+		DeleteTimeout:   30 * time.Minute,
+	}))
+
+	duration = time.Since(start)
+	klog.Infof("Destroyer.Run execution time: %v", duration)
+
+	for _, e := range destroyerEvents {
+		Expect(e.ErrorEvent.Err).To(BeNil())
+	}
+	for _, e := range destroyerEvents {
+		Expect(e.PruneEvent.Error).To(BeNil(), "PruneEvent: %v", e.PruneEvent)
+	}
+	for _, e := range destroyerEvents {
+		if e.Type == event.WaitType {
+			Expect(e.WaitEvent.Operation).To(BeElementOf(event.ReconcilePending, event.Reconciled), "WaitEvent: %v", e.WaitEvent)
+		}
+	}
+
+	// By("Verify 1000 ConfigMaps deleted")
+	// assertUnstructuredCount(ctx, c, deployment1Obj, 0)
```

Comment:
Do you need to verify the ConfigMaps and CRs are all deleted?

Reply:
Ah, good catch. I can add them, but it will make the test much longer, adding 3001 API calls for apply and then another 3001 API calls for delete, basically duplicating the work the StatusPoller does.

Reply:
Then we don't need to run this for all the objects. Maybe just randomly choose one and verify it.

Reply:
I think we actually might be able to make one LIST call with a label selector across all namespaces. Gonna try that.

Reply:
fixed.